### PR TITLE
feat(route): add 逛丢关键字搜索

### PR DESCRIPTION
**Pull request by Huzhixin00**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/guangdiu/search/q=笔记本
/guangdiu/search/q=洗发水
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [v2 Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [v2 路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [x] Documentation / 文档说明
  - [] EN / 英文文档
  - [x] CN / 中文文档
- [x] Full text / 全文获取
  - [x] Use cache / 使用缓存
- [ ] Anti-bot or rate limit / 反爬/频率限制 
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施? 
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/v2/guangdiu/maintainer.js [hunk 1]
```diff
@@ -2,4 +2,5 @@ module.exports = {
     '/rank': ['fatpandac'],
     '/cheaps/:query?': ['fatpandac'],
     '/:query?': ['Fatpandac'],
+    '/search/:query?': ['Huzhixin00'],
```
Comment (TonyRL) [inline lib/v2/guangdiu/maintainer.js:5]:
```suggestion
    '/search/:query?': ['Huzhixin00'],
    '/:query?': ['Fatpandac'],
```

File: lib/v2/guangdiu/router.js [hunk 2]
```diff
@@ -2,4 +2,5 @@ module.exports = function (router) {
     router.get('/rank', require('./rank'));
     router.get('/cheaps/:query?', require('./cheaps'));
     router.get('/:query?', require('./index'));
+    router.get('/search/:query?', require('./search'));
```
Comment (TonyRL) [inline lib/v2/guangdiu/router.js:5]:
```suggestion
    router.get('/search/:query?', require('./search'));
    router.get('/:query?', require('./index'));
```

File: lib/v2/guangdiu/search.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+module.exports = async (ctx) => {
+    const query = ctx.params.query ?? '';
+    const url = `${host}/${query ? `search.php?${query}` : ''}`;
+    const response = await got(url);
+    const $ = cheerio.load(response.data);
+    const list = $('#mainleft > div.zkcontent > div.gooditem')
+        .map((_index, item) => ({
+            title: $(item).find('a.goodname').text().trim(),
+            link: new URL($(item).find('div.iteminfoarea > h2 > a').attr('href'), host).href,
```
Comment (TonyRL) [inline lib/v2/guangdiu/search.js:15]: Please use the `href` from `a.goodname` instead

Comment (Huzhixin00) [reply lib/v2/guangdiu/search.js]: done~

**Pull request merged**